doc/README.md

```diff
@@ -184,12 +184,67 @@ In that case, please open an issue about the particular documentation convention
   }
   ```
 
-- Use [definition lists](#definition-lists) to document function arguments, and the attributes of such arguments. For example:
+- When showing inputs/outputs of any [REPL](https://en.wikipedia.org/wiki/Read%E2%80%93eval%E2%80%93print_loop), such as a shell or the Nix REPL, use a format as you'd see in the REPL, while trying to visually separate inputs from outputs.
+  This means that for a shell, you should use a format like the following:
+  ```shell
+  $ nix-build -A hello '<nixpkgs>' \
+    --option require-sigs false \
+    --option trusted-substituters file:///tmp/hello-cache \
+    --option substituters file:///tmp/hello-cache
+  /nix/store/zhl06z4lrfrkw5rp0hnjjfrgsclzvxpm-hello-2.12.1
+  ```
+  Note how the input is preceded by `$` on the first line and indented on subsequent lines, and how the output is provided as you'd see on the shell.
+
+  For the Nix REPL, you should use a format like the following:
+  ```shell
+  nix-repl> builtins.attrNames { a = 1; b = 2; }
+  [ "a" "b" ]
+  ```
+  Note how the input is preceded by `nix-repl>` and the output is provided as you'd see on the Nix REPL.
+
+- When documenting functions or anything that has inputs/outputs and example usage, use nested headings to clearly separate inputs, outputs, and examples.
+  Keep examples as the last nested heading, and link to the examples wherever applicable in the documentation.
+
+  The content of the nested headings is still up to the writer.
+  The purpose of this convention is to provide a familiar structure for navigating the manual, so any reader can expect to find content related to inputs in an "inputs" heading, examples in an "examples" heading, and so on.
+  An example:
+  ```
+  ## buildImage
+
+  Some explanation about the function here.
+  Describe a particular scenario, and point to [](#ex-dockerTools-buildImage), which is an example demonstrating it.
+
+  ### Inputs
+
+  Documentation for the inputs of `buildImage`.
+  Perhaps even point to [](#ex-dockerTools-buildImage) again when talking about something specifically linked to it.
+
+  ### Passthru outputs
+
+  Documentation for any passthru outputs of `buildImage`.
+
+  ### Examples
+
+  Note that this is the last nested heading in the `buildImage` section.
+
+  :::{.example #ex-dockerTools-buildImage}
+
+  # Using `buildImage`
+
+  Example of how to use `buildImage` goes here.
+
+  :::
+  ```
+
+- Use [definition lists](#definition-lists) to document function arguments, the attributes of such arguments, or any other attribute sets in general. For example:
 
   ```markdown
   # pkgs.coolFunction
 
   Description of what `coolFunction` does.
+
```

Comment:
I do not think your changes address the question I raised. In the interest of clarity, note that:

* no statements were made regarding an already existing convention. There are a few different "local conventions". A reference was being made to [*one* of these local conventions](https://nixos.org/manual/nixpkgs/unstable/#trivial-builder-runCommand). For clarity, I'll refer to these "local conventions" as "dialects".

* no statements were made regarding personal preference for an existing documentation dialect over another. An attempt was made to express uncertainty around the selection of one dialect over others, without due diligence. 

* (this will be fixed) no statements were made regarding collating a list of existing documentation-dialects, or better yet, to link to existing research on this front.

To fix the last point:
* The `nixos-render-doc` work *might* have had such a collation (not easy to (re)find, therefore, not included for now). 
* [`nixdoc`](https://github.com/nix-community/nixdoc), which is already used for documenting items in [`lib`, generates output like so](https://nixos.org/manual/nixpkgs/unstable/#sec-functions-library-strings)

To fix potential issues around misunderstanding of colloquialisms, note that "to ditch" means:
> To [discard](https://en.wiktionary.org/wiki/discard) or [abandon](https://en.wiktionary.org/wiki/abandon).

I will stick to discard from now on for clarity.

The convention here---even though a work in progress---has the potential to influence new PRs suggesting changing existing `f :: A -> B -> C` structure to use a Input/Output header structure. This is one sense in which `f :: A -> B -> C` style structure may be discarded.

If a convention is ratified by this PR, then it would greenlight changing existing documentation differing from the approved structure. This is another sense in which existing dialects could be discarded.

Therefore, two issues are being raised: 

1. Please do not select a convention, without a summary of existing dialects, and some analysis of their pros and cons. 

2. Please mark clearly that this document is a work in progress that does not endorse any convention yet.